sys/mips/include/cpuregs.h

```diff
@@ -720,12 +720,23 @@
 #define	MIPS_OPCODE_C1		0x11
 
 /*
- * Bits defined for for the HWREna (CP0 register 7, select 0).
+ * Bits defined for the HWREna (CP0 register 7, select 0).
  */
 #define	MIPS_HWRENA_CPUNUM	(1<<0)	/* CPU number program is running on */
 #define	MIPS_HWRENA_SYNCI_STEP 	(1<<1)	/* Address step sized used with SYNCI */
 #define	MIPS_HWRENA_CC		(1<<2)	/* Hi Res cycle counter */
 #define	MIPS_HWRENA_CCRES	(1<<3)	/* Cycle counter resolution */
+/* BERI specific statcounters */
+#if defined(CPU_BERI)
+#define	MIPS_BERI_HWRENA_RESET_STATCOUNTERS     (1<<7)	/* statcounters reseting */
+#define	MIPS_BERI_HWRENA_ICACHE_STATCOUNTERS    (1<<8)	/* ICACHE statcounters */
+#define	MIPS_BERI_HWRENA_DCACHE_STATCOUNTERS    (1<<9)	/* DCACHE statcounters */
+#define	MIPS_BERI_HWRENA_L2CACHE_STATCOUNTERS   (1<<10)	/* L2CACHE statcounters */
+#define	MIPS_BERI_HWRENA_MEM_STATCOUNTERS       (1<<11)	/* memory statcounters */
+#define	MIPS_BERI_HWRENA_TAGCACHE_STATCOUNTERS  (1<<12)	/* tag cache statcounters */
+#define	MIPS_BERI_HWRENA_L2MASTER_STATCOUNTERS  (1<<13)	/* L2 cache master statcounters */
+#define	MIPS_BERI_HWRENA_TAGMASTER_STATCOUNTERS (1<<14)	/* tag cache master statcounters */
```

Comment:
I think I'd define them as `MIPS_HWRENA_BERI_` rather than `MIPS_BERI_HWRENA_`.

To simplify the other code, I'd also define a `MIPS_HWRENA_BERI_MASK` which is either the OR or all of them or 0 depending on `CPU_BERI`.